api/client/commands.go

```diff
@@ -222,6 +251,9 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 	if *pull {
 		v.Set("pull", "1")
 	}
+
+	v.Set("f", *dockerfileName)
```

Comment:
Ah, well, the original PR I submitted used "-d" instead but at some point people seemed to gravitate towards -f (perhaps because 'make' uses -f to point to the makefile?) so I switched it.  I do think that if we use -d on the API then we should probably use it on the cmd line too. 
I'm ok with either, but would need the appropriate UX approval first I think.